File: FullStackExampleProject/Models/ToDo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace FullStackExampleProject.Models
{
    public class ToDo
    {
        [Key]
        public int Id { get; set; }
        public int AssignedTo { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int HoursNeeded { get; set; }
        public bool IsCompleted { get; set; }
    }
}

File: FullStackExampleProject/ClientApp/src/app/ToDo.ts
export interface ToDo{
    id: number;
    assignedTo: number;
    name: string;
    description: string;
    hoursNeeded: number;
    isCompleted: boolean;
}

// Converts JSON strings to/from your types
export class ConvertToDo {
    public static toToDo(json: string): ToDo {
        return JSON.parse(json);
    }

    public static toDotoJson(value: ToDo): string {
        return JSON.stringify(value);
    }
}